### PR TITLE
Update IoT Hub sample to use new Events Hubs API

### PR DESCRIPTION
**Pull request by srnagar**
## Purpose

The current sample to demonstrate reading events from an Event Hub that published from an IoT Hub device uses an older version of the SDK. This PR updates the sample to use the new version of Event Hubs SDK (azure-messaging-eventhubs version 5.1.0).

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Pull Request Type
What kind of change does this Pull Request introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [x] Other: Update sample to use latest version of Event Hubs SDK

## How to Test
*  Get the code
```
git clone https://github.com/srnagar/azure-iot-samples-java
cd azure-iot-samples-java
git checkout eh-iot-sample
mvn clean install
```

* Test the code

```
Add your event hub compatible endpoint information by running Azure CLI, and run the code in ReadDeviceToCloudMessages.java. Send data to your IotHub via any means you are familiar with in another program.
```

## What to Check
Verify that the sample prints out messages that you have sent

### CODE REVIEW

**Comment by ramya-rao-a** [comment]
cc @elhorton, @jebrando, @JimacoMS3, @robinsh, @philmea

This is similar to what was done to the Node sample in https://github.com/Azure-Samples/azure-iot-samples-node/pull/39

**Comment by JimacoMS3** [comment]
@ramya-rao-a Thanks! This looks great! 

Just a couple of questions:

1. The EH compatible connection string variables have been changed from the names in the [quickstart documentation](https://docs.microsoft.com/en-us/azure/iot-hub/quickstart-send-telemetry-java#read-the-telemetry-from-your-hub) (step 2). Is there a reason for this (i.e. is it more java-like)? Let me know whether I should update the doc. 

2. Does the screenshot in the [doc](https://docs.microsoft.com/en-us/azure/iot-hub/quickstart-send-telemetry-java#read-the-telemetry-from-your-hub)  (step 4) need to be updated? (I'm OK with the date being off.)

Thanks,

Jimaco

**Comment by srnagar** [comment]
1. The variable names are more java-like for static final constants. It would be great to update the document to reflect these new names. Let me know if that's not the case, I can change the names back to what it previously was.

2. The final outcome is going to very similar to the one in the screenshot with just one difference. 
Instead of `Telemetry received`, users would see `Telemetry received from partition 0`. So, if it's too difficult to update the screenshot, leaving it as-is will be okay.

**Comment by JimacoMS3** [comment]
Thanks @srnagar . I created this PR to update the constant names: https://github.com/MicrosoftDocs/azure-docs-pr/pull/115302 

I'm OK publishing with or without an updated screenshot. I suggest, however, that we wait until after /BUILD starts on 5/19. We normally publish docs 2x/day at roughly 10-10:30 AM and 3-3:30 AM. With the ramup-up to /BUILD there are lock-downs in place and changes to the pub schedule (I'm not sure of the precise details). Things open back up again on the morning of 5/19. We will need to collaborate somewhat and target one of the publishing times at that point. (We don't have to be exact, but I want to avoid a situation where the sample gets updated, but the docs don't update for a few days.)  In the interim, if I get time, I'll grab your code and try to update the screenshot.

Sound good?

**Comment by srnagar** [comment]
Sounds good. We can wait until after /BUILD is done to get this merged. Thanks for the update. Let you know if you have any issues updating the screenshot.

**Comment by JimacoMS3** [comment]
@srnagar Great. I'll let you know if I have any trouble with the screenshot.

**Comment by JimacoMS3** [comment]
@srnagar I just cloned your fork and ran this. The output is a bit smooshed vertically:

image

Can you add line feeds so that it looks more like what is [currently live](https://docs.microsoft.com/en-us/azure/iot-hub/quickstart-send-telemetry-java#read-the-telemetry-from-your-hub) (scroll to step 4)? I just find it a bit easier to parse as a human :),

**Comment by srnagar** [comment]
@JimacoMS3 Thanks for the feedback. I have updated the PR to include line breaks.

**Comment by JimacoMS3** [comment]
@srnagar We're almost there :). 

1. Please make sure there is a **blank line** between reads. Right now they all run together:

    image

2. Can we set the handler to read from the current time by default? (Otherwise it splats out data from past runs, etc. -- might be confusing for a beginner.)

3. Can we make sure that those experiences will be consistent across the handlers exposed in the sample?

**Comment by srnagar** [comment]
@JimacoMS3 Update the PR to fix new lines and read latest events by default.

**Comment by JimacoMS3** [comment]
@srnagar Thanks -- almost! Need a blank line between reads. 

image

**Comment by srnagar** [comment]
@JimacoMS3 - done!

**Comment by JimacoMS3** [comment]
@srnagar Thanks! Looks great!

**Comment by JimacoMS3** [comment]
@srnagar, @ramya-rao-a Any objections to merging this sample around 3 pm today? (@robinsh can help me with it if neither of you have merge privileges.)

**Comment by srnagar** [comment]
Please go ahead and merge! Thank you!

File: iot-hub/Quickstarts/read-d2c-messages/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+Azure Storage Blob to create checkpoints](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/eventhubs/azure-messaging-eventhubs-checkpointstore-blob/src/samples/java/com/azure/messaging/eventhubs/checkpointstore/blob/EventProcessorBlobCheckpointStoreSample.java).
+
+Note that this requires adding a new dependency in your `pom.xml` to use the [Azure Storage Blob checkpoint store](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/eventhubs/azure-messaging-eventhubs-checkpointstore-blob/README.md).
+
+## WebSocket and proxy
+
+If using WebSockets, configure the `EventHubClientBuilder` to use transport type `AmqpTransportType.AMQP_WEB_SOCKETS`.
+
+If your application runs behind a proxy server, then you need to configure the proxy options and set the transport 
+type to `AmqpTransportType.AMQP_WEB_SOCKETS` as shown in [the sample](./src/main/java/com/microsoft/docs/iothub/samples/ReadDeviceToCloudMessages.java).
```
Comment (ramya-rao-a) [inline iot-hub/Quickstarts/read-d2c-messages/README.md:35]: I'd recommend re-structuring as below to put emphasis on proxy options

```suggestion
If your application runs behind a proxy server, then apart from setting the transport type to `AmqpTransportType.AMQP_WEB_SOCKETS`,
you need to configure the proxy options as shown in [the sample](./src/main/java/com/microsoft/docs/iothub/samples/ReadDeviceToCloudMessages.java). 
```

**Comment by ramya-rao-a** [comment]
Looks good to me, just a tiny suggestion in https://github.com/Azure-Samples/azure-iot-samples-java/pull/15/files#r430656512

**Comment by robinsh** [comment]
If we’re going to merge this, let’s not wait until the last minute and hope it gets added in. Let’s just merge it. It won’t be published until 3:00. What do you think, Jimaco? Can you make this last suggestion from Ramya first?

File: iot-hub/Quickstarts/read-d2c-messages/README.md (hunk 1 above)
Comment (JimacoMS3) [reply iot-hub/Quickstarts/read-d2c-messages/README.md:35]: Hey Ramya -- I think 'apart' is confusing. Do you mean this:

If your application runs behind a proxy server, then, in addition to setting the transport type to `AmqpTransportType.AMQP_WEB_SOCKETS`, you also need to configure the proxy options as shown in the `setupProxy` method in [the sample](./src/main/java/com/microsoft/docs/iothub/samples/ReadDeviceToCloudMessages.java).

Comment (JimacoMS3) [reply iot-hub/Quickstarts/read-d2c-messages/README.md:35]: @srnagar

**Comment by JimacoMS3** [comment]
@robinsh I added a comment for Ramya to clarify. I was hoping the owner could update the PR. I'm not overly concerned with whether we're a couple hours early or a couple hours late on merging this. I'll go ahead and sign-off on the doc update to make sure it gets published.

**Comment by robinsh** [comment]
You can wait until you get the last change in there. Better than merging twice. 😝

**Comment by srnagar** [comment]
@JimacoMS3 and @ramya-rao-a I have updated the readme text as suggested.

File: iot-hub/Quickstarts/read-d2c-messages/README.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+Azure Storage Blob to create checkpoints](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/eventhubs/azure-messaging-eventhubs-checkpointstore-blob/src/samples/java/com/azure/messaging/eventhubs/checkpointstore/blob/EventProcessorBlobCheckpointStoreSample.java).
+
+Note that this requires adding a new dependency in your `pom.xml` to use the [Azure Storage Blob checkpoint store](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/eventhubs/azure-messaging-eventhubs-checkpointstore-blob/README.md).
+
+## WebSocket and proxy
+
+If using WebSockets, configure the `EventHubClientBuilder` to use transport type `AmqpTransportType.AMQP_WEB_SOCKETS`.
+
+If your application runs behind a proxy server, then, in addition to setting the transport type to 
+AmqpTransportType.AMQP_WEB_SOCKETS, you also need to configure the proxy options as shown in the setupProxy method in
```
Comment (JimacoMS3) [inline iot-hub/Quickstarts/read-d2c-messages/README.md:35]: @srnagar  Should have tics ( ` ) around  AmqpTransportType.AMQP_WEB_SOCKETS and setupProxy to give code formatting.  

(PS: I'd do this myself, but I haven't really messed much with making changes in other people's forks and don't want to stomp all over your stuff)

Comment (srnagar) [reply iot-hub/Quickstarts/read-d2c-messages/README.md]: Done

Comment (JimacoMS3) [reply iot-hub/Quickstarts/read-d2c-messages/README.md]: Awesome! Thanks!

**Comment by JimacoMS3** [comment]
@sign-off

File: iot-hub/Quickstarts/read-d2c-messages/README.md (hunk 1 above)
Comment (ramya-rao-a) [reply iot-hub/Quickstarts/read-d2c-messages/README.md]: yes, that sounds much better :)

**Comment by robinsh** [comment]
#sing-off doesn't work either.

**Comment by JimacoMS3** [comment]
#sign-off

**Comment by robinsh** [comment]
okay, I'll merge. Have a nice day, Jicamo. I mean, Jimaco.

**Pull request merged**